pdc/apps/release/serializers.py

```diff
@@ -15,7 +15,27 @@
                      validateCPE)
 from . import signals
 from pdc.apps.common.models import SigKey
-from pdc.apps.repository.models import Service
+from pdc.apps.repository.models import PushTarget, Service
+
+
+def convert_push_targets_to_mask(data, instance, parent_key):
+    """
+    Converts "allowed_push_targets" in data to "masked_push_targets" (depending on data in parent).
+    """
+    if 'allowed_push_targets' in data:
+        parent = data.get(parent_key)
+        if parent is None and instance is not None:
+            parent = getattr(instance, parent_key)
+
+        allowed_push_targets = set(data.pop('allowed_push_targets'))
+        available_push_targets = set(parent.allowed_push_targets.all()) if parent else set()
+        invalid_push_targets = [
+            push_target.name for push_target in allowed_push_targets - available_push_targets]
+        if invalid_push_targets:
+            raise serializers.ValidationError(
+                {'detail': 'Invalid push targets: %s' % invalid_push_targets})
+
```

Comment:
I think we should give more info here, like "targets should be the sub-set of the product".

Reply:
How about following?

```bash
> pdc_client -r releases/rhel-7.1 -x PATCH -d '{"allowed_push_targets":["rhn-qa"]}'  
ERROR:pdc_client:400 {"detail":["Push targets must be allowed in parent product_version: [u'rhn-qa']"]}
> pdc_client -r product-versions/rhel-7 -x PATCH -d '{"allowed_push_targets":["rhn-qa"]}' 
ERROR:pdc_client:400 {"detail":["Push targets must be allowed in parent product: [u'rhn-qa']"]}
```

Reply:
cool, more clear now.